framework-cloud/framework-openshift-strimzi/src/main/java/org/kie/cloud/strimzi/TopicOperator.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.cloud.strimzi;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.io.FileUtils;
+import org.kie.cloud.api.scenario.HACepScenario;
+import org.kie.cloud.openshift.resource.Project;
+import org.kie.cloud.openshift.resource.impl.ProjectImpl;
+
+public class TopicOperator {
+    private final String DEFAULT_TOPIC_NAME = "my-topic";
+    private final String KAFKA_TOPIC_EXAMPLE = "examples/topic/kafka-topic.yaml";
+
+    private HACepScenario haCepScenario;
+
+    public TopicOperator(final HACepScenario haCepScenario) {
+        this.haCepScenario = haCepScenario;
+    }
+
+    public void createTopic(final String name) {
+        final File topicYamlFile = new File(haCepScenario.getAMQStreamsDirectory(), KAFKA_TOPIC_EXAMPLE);
+        String topicYaml;
+        try {
+            topicYaml = FileUtils.readFileToString(topicYamlFile, "UTF-8");
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to read topic YAML file: " + topicYamlFile.getAbsolutePath(), e);
+        }
+        topicYaml = topicYaml.replace(DEFAULT_TOPIC_NAME, name);
+
+        final File tmpYamlFile;
+        try {
+            tmpYamlFile = File.createTempFile("topic-", ".yaml");
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to create temp file for topic", e);
+        }
+
+        try {
+            FileUtils.writeStringToFile(tmpYamlFile, topicYaml, "UTF-8");
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to write topic YAML", e);
+        }
+
+        final Project project = new ProjectImpl(haCepScenario.getNamespace());
```

Comment:
Project implements AutoCloseable so it needs to be closed. From my perspective it would be better to pass it in constructor instead of HACepScenario, to decouple TopicOperator from specific scenario implementation. Also, the project is managed in scenario (being closed when scenario is undeployed).